src/components/AddSenior.tsx

```diff
@@ -114,28 +121,42 @@ const AddSenior = ({
   seniorPatch,
   setSeniorPatch,
 }: AddSeniorProps) => {
-  const [seniorData, setSeniorData] = useState<SeniorData>(EMPTY_SENIOR);
   const [selectedStudents, setSelectedStudents] = useState<User[]>([]);
   const [currentImage, setCurrentImage] = useState<string | StaticImageData>(
     ImageIcon
   );
   const [confirm, setConfirm] = useState<boolean>(false);
   const [error, setError] = useState<boolean>(false);
+  const [edited, setEdited] = useState<boolean>(false);
+
+  const { register, handleSubmit, reset, setValue } = useForm<SeniorData>({
+    resolver: zodResolver(seniorFormSchema),
+  });
+
+  const onSubmit: SubmitHandler<SeniorData> = async (data, event) => {
```

Comment:
I think the data needs to validated before making the API request.

Reply:
Data is automatically validated because we pass in the zodResolver function to the useForm hook.